reliability-engineering/terraform/modules/concourse-monitoring/lb.tf

```
@@ -0,0 +1,103 @@
+resource "aws_lb" "concourse_monitoring" {
+  name               = "${var.deployment}-concourse-monitoring"
+  internal           = false
+  load_balancer_type = "application"
+  security_groups    = ["${aws_security_group.concourse_monitoring_lb.id}"]
+  subnets            = ["${var.public_subnet_ids}"]
+
+  tags = {
+    Name       = "${var.deployment}-concourse-monitoring"
+    Deployment = "${var.deployment}"
+  }
+}
+
+resource "aws_lb_listener" "concourse_monitoring_https" {
+  load_balancer_arn = "${aws_lb.concourse_monitoring.arn}"
+  port              = "443"
+  protocol          = "HTTPS"
+  ssl_policy        = "ELBSecurityPolicy-2015-05"
+  certificate_arn   = "${aws_acm_certificate.concourse_monitoring.arn}"
```

```suggestion
  certificate_arn   = "${aws_acm_certificate_validation.concourse_web.certificate_arn}"
```
I don't see much point in having the `aws_acm_certificate_validation` resource unless you get the arn from it like this.  This is what actually causes terraform to wait for validation before attaching it to the load balancer.

Note that this is much more important when making changes to infrastructure (eg changing the certificate) than in the initial buildout.

---

I use the resource to generate the DNS records. This is a good change though.

---

> I use the resource to generate the DNS records. 

but that's... not what `aws_acm_certificate_validation` does? And you still create the DNS records yourself anyway? 😕